internal/k8s/reconciler/route.go

```diff
@@ -353,7 +195,7 @@ func (r *K8sRoute) Validate(ctx context.Context) error {
 					nsName := getNamespacedName(ref.Name, ref.Namespace, route.Namespace)
 					msg := fmt.Sprintf("Cross-namespace routing not allowed without matching ReferenceGrant for Service %q", nsName)
 					r.logger.Warn("Cross-namespace routing not allowed without matching ReferenceGrant", "refName", nsName.Name, "refNamespace", nsName.Namespace)
-					r.resolutionErrors.Add(service.NewRefNotPermittedError(msg))
+					r.RouteState.ResolutionErrors.Add(service.NewRefNotPermittedError(msg))
```

Comment:
it'll be so nice when we get to the validator and other service structs later -- these multi-chained function calls are gross 😢